bodhi/server/models.py

```diff
@@ -3253,6 +3302,204 @@ def __json__(self, request=None, anonymize=False):
+        new Compose objects that match those groupings.
+
+        Note that calling this will cause each of the updates to become locked once the transaction
+        is committed.
+
+        Args:
+            updates (list): A list of :class:`Updates <Update>` that you wish to Compose.
+        Returns:
+            list: A list of new compose objects for the given updates.
+        """
+        work = {}
+        for update in updates:
+            if not update.request:
+                log.info('%s request was revoked', update.title)
+                continue
+            # ASSUMPTION: For now, updates can only be of a single type.
+            ctype = None
+            for build in update.builds:
+                if ctype is None:
+                    ctype = build.type
+                elif ctype is not build.type:  # pragma: no cover
+                    # This branch is not covered because the Update.validate_builds validator
+                    # catches the same assumption breakage. This check here is extra for the
+                    # time when someone adds multitype updates and forgets to update this.
+                    raise ValueError('Builds of multiple types found in %s'
+                                     % update.title)
+            # This key is just to insert things in the same place in the "work"
+            # dict.
+            key = '%s-%s' % (update.release.name, update.request.value)
+            if key not in work:
+                work[key] = cls(request=update.request, release_id=update.release.id,
+                                release=update.release)
+            # Lock the Update. This implicity adds it to the Compose because the Update.compose
+            # relationship joins on the Compose's compound pk for locked Updates.
+            update.locked = True
+
+        return work.values()
+
+    @property
+    def security(self):
+        """
+        Return whether this compose is a security related compose or not.
+
+        Returns:
+            bool: ``True`` if any of the :class:`Updates <Update>` in this compose are marked as
+                security updates.
+        """
+        for update in self.updates:
+            if update.type is UpdateType.security:
+                return True
+        return False
+
+    @staticmethod
+    def update_state_date(target, value, old, initiator):
+        """
+        Update the ``state_date`` when the state changes.
+
+        Args:
+            target (Compose): The compose that has had a change to its state attribute.
+            value (EnumSymbol): The new value of the state.
+            old (EnumSymbol): The old value of the state
+            initiator (sqlalchemy.orm.attributes.Event): The event object that is initiating this
+                transition.
+        """
+        if value != old:
+            target.state_date = datetime.utcnow()
+
+    def __eq__(self, other):
```

Comment:
I am honestly not sure that this is a good idea.
Since this might confuse people a lot, since normally `==` would say whether two objects are the actual same compose.

Reply:
I had thought this would be important for the sorting to work, but I now believe that I only need the ```__le__()``` to do that. I'll drop this.